add invite email in server

GuodongH opened the pull request.

Harukine commented on server/controller/invite.js:
```diff
@@ -0,0 +1,35 @@
+const User = require('../models/User');
+
+let errors = {};
+errors.msg = '';
```
initializing errors.msg to empty is not needed, please remove line 8

Harukine commented on server/controller/invite.js:
```diff
@@ -0,0 +1,35 @@
+errors.msg = '';
+
+sgMail.setApiKey(
+  process.env.SENDGRID_API_KEY ||
```
need to move this to configs

Harukine reviewed (approved):
Just need to add "@sendgrid/mail": "^6.4.0", to the dependencies in package.json
and I think creating a new route called 'invites' would be better than putting it in the users route, 
there will be a email and link paths

GuodongH committed:
update invite by email feature
- move invites router to new js file (thank you @Jacob
- add dependencies in package.json
- add get current user in user route for debug use

Harukine commented on server/routes/api/users.js:
```diff
@@ -1,109 +1,128 @@
+// @route   GET api/users/current
+// @desc    Return current user
+// @access  Private
+router.get(
```
This does not need to be committed into the repo
